### ISSUE TITLE
Allow user to specify log base

### ISSUE THREAD

**Issue by BrandonHaynes** [uwescience/raco#393]
I have a `while` statement that needs to run log_2 n times.  If overloading is supported, it would be nice to be able to specify a base for the existing `log` function.  (I guess a `log2` function would work as well; I find both options to be more clear than `log(n) / log(2)`).

**Comment by dhalperi** [comment]
You know that you can define little inline functions (like Lambda expressions), right?

```myrial
def logN(a, b): log(a) / log(b);
```

will let you use `logN` in your program.

**Comment by dhalperi** [comment]
But, yes, this is a great suggestion. The place to do this is in the Expression Library: https://github.com/uwescience/raco/blob/master/raco/expression/expressions_library.py#L34

Overloading works (see `SafeDiv`), but I am not certain what will happen if you add a 2-ary `log` to the expressions library while leaving the unary `log`, which is in the language, alone... :) It might just work!

**Comment by BrandonHaynes** [comment]
Oh nice -- I did not know that the language supported inline functions.  Thanks for mentioning that!

**Comment by domoritz** [comment]
We need docs on that. And how to write stateful apply and UDAs. I'd be happy to write the latter some time next month. Ref https://github.com/uwescience/myria-docs/blob/master/myriaql.markdown

**Comment by bmyerz** [comment]
@domoritz is there a issue in myria-docs to track that?

**Comment by domoritz** [comment]
@bmyerz No, oh I mean yes https://github.com/uwescience/myria-docs/issues/9
